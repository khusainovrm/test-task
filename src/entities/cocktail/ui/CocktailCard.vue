<template>
  <div class="p-2 flex flex-col gap-4 sm:gap-2 sm:flex-row">
    <!--    content-->
    <div class="flex-1 order-2 sm:order-1">
      <h1 class="mb-2">{{ drink.strDrink }}</h1>

      <section class="mb-8">
        <p>{{ drink.strCategory }}</p>
        <p>{{ drink.strAlcoholic }}</p>
        <p>{{ drink.strGlass }}</p>
      </section>

      <section class="mb-8">
        <h3>Instructions:</h3>
        <p>{{ drink.strInstructions }}</p>
      </section>

      <section>
        <h2>List of ingredients:</h2>
        <p v-for="item in ingredients" :key="item">{{ item }}</p>
      </section>
    </div>

    <!--    image-->
    <div class="flex-1 order-1 sm:order-2">
      <ImageLazy :src="drink.strDrinkThumb!" class="rounded-2xl aspect-[4/3] object-cover" alt="drink" />
    </div>
  </div>
</template>

<script setup lang="ts">
import ImageLazy from '@/shared/ui/image/ImageLazy.vue';
import type { Cocktail } from '../model';
import { useCocktailDetailed } from '../model';

const props = defineProps<{
  drink: Cocktail;
}>();

const { ingredients } = useCocktailDetailed(props.drink);
</script>
