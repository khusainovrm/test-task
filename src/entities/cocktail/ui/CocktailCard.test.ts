import { describe, expect, test } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import CocktailCard from './CocktailCard.vue';

describe('CocktailCard', () => {
  test('mount component', async () => {
    expect(CocktailCard).toBeTruthy();

    const wrapper = shallowMount(CocktailCard, {
      props: {
        drink: {
          idDrink: '1',
          strDrink: 'Vino',
          strInstructions: 'mix',
        },
      },
    });

    expect(wrapper.text()).toContain('Instructions:');
    expect(wrapper.text()).toContain('List of ingredients:');
    expect(wrapper.text()).toContain('Vino');
  });
});
