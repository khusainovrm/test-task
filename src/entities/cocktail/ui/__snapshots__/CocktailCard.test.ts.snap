// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CocktailCard > mount component 1`] = `
"<div class="p-2 flex flex-col gap-4 sm:gap-2 sm:flex-row">
  <!--    content-->
  <div class="flex-1 order-2 sm:order-1">
    <h1 class="mb-2">Vino</h1>
    <section class="mb-8">
      <p></p>
      <p></p>
      <p></p>
    </section>
    <section class="mb-8">
      <h3>Instructions:</h3>
      <p></p>
    </section>
    <section>
      <h2>List of ingredients:</h2>
    </section>
  </div><!--    image-->
  <div class="flex-1 order-1 sm:order-2">
    <!--v-if-->
  </div>
</div>"
`;
